### ISSUE TITLE
IPv4 connectivity to ndt pods is broken.

### ISSUE THREAD

**Issue by nkinkade** [m-lab/k8s-support#164]
Attempting to connect to an NDT pod's IPv4 address on ports 3001 or 3010 fails. Interestingly, connecting to the IPv6 address of the pod works.This somewhat points to a routing or iptable issues.

For example, this fails:

$ nc -z ndt-iupui-mlab4**v4**-den04.measurement-lab.org 3010

.... but this works:

$ nc -z ndt-iupui-mlab4**v6**-den04.measurement-lab.org 3010

**Comment by nkinkade** [comment]
This issue is resolved with PR #166. Once that PR is merged and fully tested in sandbox this issue can be closed.

**Comment by nkinkade** [comment]
PR merged and issues seems resolved in sandbox. Closing.

**Comment by nkinkade** [comment]
It turns out this is not fixed, and I'm very confused as to why my and @pboothe's tests with the changes in PR #166 worked. 

It turns out that "l3" mode for the ipvlan driver requires all subinterfaces to be on different subnetworks than the "master" interface. This is not our use can, and it cannot really work this way easily.

**Comment by nkinkade** [comment]
This is fixed again with PR #167. The result is that experiment pods will not have access to kubernetes  cluster services, but all other pods will. Experiment pods don't currently use k8s services, so this is not an issue.

**Comment by nkinkade** [comment]
Reopening this issue, yet again, because this problem, in one form or another still exists.  Now the issue is conflicting pod IP addresses. Rather the reiterate everything I already put in Slack, I'm going to just paste my Slack findings here. I believe I understand the problem now, but how to get around it is another question.

```
Nathan [17:03]
 Duplicate pod IPs could be some sort of race condition with flannel:

```cadvisor-lg9tq                       1/1     Running             0          104m    192.168.6.195     mlab4.den04.measurement-lab.org     <none>           <none>
ndt-4qfcj                            4/4     Running             3          104m    192.168.6.195     mlab4.den04.measurement-lab.org     <none>           <none>```
The uptime for both of those pods is 104m, and they have the same IP. Maybe this means that pods shouldn't be created at nearly identical times?
Nathan [17:22]
 Ah:
> Flannel doesn't assign IP addresses to pods. Maybe you have a problem with [...] IPAM plugin
In this case it would be the bridge plugin.
Nathan [17:52]
 I think I see what is happening now. We now have two flannel configurations: `flannel-conf` and `flannel-experiment-config`. The `host-local` IPAM plugin that flannel uses stores IP allocations in `/var/lib/cni/networks/`. Each "network" (equivalent, apparently, to separate flannel configs) is different. Hence we get:
```bash-4.3# find /var/lib/cni/networks/                          
/var/lib/cni/networks/
/var/lib/cni/networks/flannel-conf
/var/lib/cni/networks/flannel-conf/192.168.6.196
/var/lib/cni/networks/flannel-conf/last_reserved_ip.0
/var/lib/cni/networks/flannel-conf/lock
/var/lib/cni/networks/flannel-experiment-conf
/var/lib/cni/networks/flannel-experiment-conf/192.168.6.196
/var/lib/cni/networks/flannel-experiment-conf/last_reserved_ip.0
/var/lib/cni/networks/flannel-experiment-conf/lock```
I'm guessing that when the host-local ipam plugin for the `flannel-experiment-conf` config goes to see whether 192.168.6.196 is used, it doesn't find it and uses it... .when in reality it is already assigned to a pod in the `flannel-conf` network.
```

**Comment by nkinkade** [comment]
I believe this should be fixed now with PR m-lab/epoxy-images#100. The fix looks fine in mlab-sandbox, and on a spot check of mlab4.den04, and I am currently rebooting all mlab4 nodes to pick up this change. I'm going to leave this open over the weekend to be sure no issues arise. If all looks good on Monday, then I will close this.
